File: site/pages/stats/DPSAlgorithm.js
import Algorithm from './Algorithm'
import {POKEMON_LIST} from "app/data/Database";

class DPSAlgorithm extends Algorithm {
  name = "dps";

  compute(pokemon, quickMove, chargeMove, typeEffectiveness) {
    const moveDps = this.calcMoveDps(pokemon, quickMove, chargeMove, typeEffectiveness);
    return pokemon.getAttack() * moveDps
  }
}

const inst = new DPSAlgorithm();
inst.rank(POKEMON_LIST);
export default inst

File: src/com/sucy/go/ranks/impl/DPSAlgorithm.java
package com.sucy.go.ranks.impl;

import com.sucy.go.Move;
import com.sucy.go.Pokemon;
import com.sucy.go.ranks.RankAlgorithm;

/**
 * PokemonGo © 2018
 * com.sucy.go.ranks.impl.DPSAlgorithm
 */
public class DPSAlgorithm extends RankAlgorithm {
    @Override
    public String getName() {
        return "dps";
    }

    @Override
    public int compute(
            final Pokemon pokemon, final Move quickMove, final Move chargeMove) {

        float moveDps = calcMoveDps(pokemon, quickMove, chargeMove);
        return (int)(10000 * pokemon.attack() * moveDps);
    }
}
